modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -1248,25 +1264,27 @@ void start(UUID localNodeId) {
         }
 
         private CompletableFuture<Boolean> onPrimaryElected(PrimaryReplicaEventParameters parameters) {
-            ReplicationGroupId groupId = parameters.groupId();
-            ReplicaStateContext context = getContext(groupId);
+            TablePartitionId replicationGroupId = (TablePartitionId) parameters.groupId();
```

Comment:
There's no need to convert abstract ReplicationGroupId to specific TablePartitionId. ReplicaManager should be unaware of the specific type.

Reply:
Done. Don't remember why I cast it